polyhost/services/fontpack_render.py

```
@@ -0,0 +1,210 @@
+    """Bounds-checked glyph lookup — a cp below font.first would otherwise index
+    negatively and silently render the wrong glyph."""
+    if cp < font.first or cp > font.last:
+        raise ValueError(f"codepoint U+{cp:04X} outside font range "
+                         f"U+{font.first:04X}..U+{font.last:04X}")
+    return font.glyphs[cp - font.first]
+
+
+def glyph_to_image(font, cp: int, scale: int = 1, fg: int = 255, bg: int = 0):
+    """Rasterise one glyph at native size (optionally nearest-scaled).
+
+    Returns an ('L', (w*scale, h*scale)) image, or a 1×1 `bg` image for a
```

Ruff (0.15.18)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)